usaspending_api/disaster/v2/views/agency/spending.py

```diff
@@ -169,9 +169,18 @@ def total_queryset(self):
                 Sum(
                     Case(
                         When(
-                            self.final_period_submission_query_filters,
-                            then=F("obligations_incurred_by_program_object_class_cpe")
-                            + F("deobligations_recoveries_refund_pri_program_object_class_cpe"),
+                            self.final_period_submission_query_filters & Q(prior_year_adjustment="B"),
+                            then=self.file_b_obligation_columns["pya_b"],
+                        ),
+                        When(
+                            self.final_period_submission_query_filters & Q(prior_year_adjustment="P"),
+                            then=self.file_b_obligation_columns["pya_p"],
+                        ),
+                        When(
+                            self.final_period_submission_query_filters
+                            & Q(Q(prior_year_adjustment="X") | Q(prior_year_adjustment__isnull=True)),
+                            then=F(self.file_b_obligation_columns["base_pya_x"])
+                            + self.file_b_obligation_columns["pya_x"],
                         ),
                         default=Value(0),
                         output_field=DecimalField(max_digits=23, decimal_places=2),
```

Comment:
This is a nit and outside of the scope of changes but we may not need to provide defaults within the `Case` blocks since it is handled in the outer `Coalesce` blocks?
```suggestion
```

Reply:
Good call. This seems to be a pattern we use in a few different places so I am not quite sure if there is a reason or if we are good to remove it. Given to your point that it is out of the scope of this PR I am going to leave this but will take a look and see if we could clean this up in the future.